src/components/toggle-panel.tsx

```diff
@@ -0,0 +1,40 @@
+import { inject, observer } from "mobx-react";
+import * as React from "react";
+import { BaseComponent, IBaseProps } from "./base";
+
+import "./toggle-panel.sass";
+import { SwitchComponent } from "./switch";
+
+interface IProps extends IBaseProps {
+  title: string;
+  switchOn: boolean;
+  buttonText: string;
+  label: string;
+  buttonClick?: () => void;
+}
+interface IState {}
+
+@inject("stores")
+@observer
+export class TogglePanelComponent extends BaseComponent<IProps, IState> {
```

Comment:
I might just call this a `LabelledSwitchComponent` or `SwitchPanelComponent`, as-is it's not obvious to me why one is called a `Switch` and one is called a `Toggle`

Reply:
I'm fine with any names that make sense and are easy to parse when we revisit the project.  It's a little more than a labelled switch.  I do like `SwitchPanelComponent` - let's go with that.

Reply:
Fixed.